### PR TITLE
UTIL-1583 MyAccount - Salesflow - offer purchase form

### PR DESCRIPTION
**Pull request by bmxmaga**
https://berlingskemedia.atlassian.net/browse/UTIL-1583

### CODE REVIEW

File: backend/src/utils/http.ts [hunk 1]
```diff
@@ -59,18 +61,31 @@ export class Http {
...
+            fulfill(result);
           }
         }
       });
     });
   }
 
+  private getTransformedBody(body: any): any {
+    if (this.transform) {
+      return JSON.parse(body);
```
Comment (bmmkoc) [inline backend/src/utils/http.ts]: `JSON.parse` probably needs a try/catch block here.

File: d/kundeunivers.d.ts [hunk 2]
```diff
@@ -108,3 +108,138 @@ export interface SuspendOrderResponse {
 export interface RemoveOrderResponse {
   removed_sap_order_id?: string;
 }
+
+export interface Offer {
```
Comment (bmmkoc) [inline d/kundeunivers.d.ts]: o_O

Comment (bmmkoc) [inline d/kundeunivers.d.ts]: Aren't interfaces normally for functions only? What's the benefit of these kind of definitions? Just asking :-)

File: src/app/purchase/offer/offer.component.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      phone: ['', [Validators.required, Validators.minLength(8), Validators.maxLength(8)]],
+      terms: [false, Validators.required],
+      offer_id: ['', Validators.required]
+    });
+  }
+
+  parseAddress(address: string): any {
+    const [street, addressParts] = this.parseStreetName(address);
+    const addressPartsMatch: RegExpMatchArray = addressParts
+      .match(/\u0020?([0-9]{0,3})\u0020?([A-Z1-9])?\u0020?([0-9]{0,2})?\u0020?([0-9]{0,4}|[A-Z]{1,4})?/i);
```
Comment (bmmkoc) [inline src/app/purchase/offer/offer.component.ts]: Is this flexible enough? Is it combined with frontend validation to limit the input, or could `addressParts` by anything?

File: backend/src/gigya/gigya.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    return {
+      ...params,
+      apiKey: process.env.GIGYA_API_KEY,
+      secret: process.env.GIGYA_SECRET
+    };
+  }
+
+  public static findUser(email: string): Promise<RichResult<FindUserResponse>> {
+    return Gigya.http.get('/accounts.search', Gigya.getParams({
+      query: `SELECT UID from accounts WHERE profile.email="${email}"`
```
Comment (bmxmaga) [inline backend/src/gigya/gigya.ts]: This is not SQL. This is Gigya's SQL'ish syntax ;)
And, yes, they say it's SQL injection safe.

File: src/app/purchase/offer/offer.component.ts (hunk 3 above)
Comment (bmxmaga) [inline src/app/purchase/offer/offer.component.ts]: I will definitely add some unit tests for this but for now I just tested it manually.
I wanted to deliver it for the deadline. I hope that it's ok?

**Pull request merged**